### PR TITLE
Fix leaking hocon objects due to multiple fallback injection in ActorMaterializer

### LINKED ISSUE #4659: Memory Leak when using many short lived instances of ActorMaterializer
- Which Akka.Net version you are using: 1.4.x
- On which platform you are using Akka.Net: Windows 10

I wonder wether it is allowed to create and dispose of many instances of ActorMaterializers.
I have an actor that is created many times, and that actor starts a short lived akka stream, needing a ActorMaterializer.
So I thought it could be nice, if that actor just creates an ActorMaterializer and when it's done disposes it.
That does not seem to be a good idea, as each instance of an ActorMaterializer seems to create many other objects, that do not get cleaned up, once the ActorMaterializer gets disposed.
You can easily reproduce it with the following code:

```
ActorSystem actorSystem = ActorSystem.Create("myActorSystem");
Console.ReadKey(true);
while (true)
{
  ActorMaterializer materializer = actorSystem.Materializer();
  materializer.Shutdown();
  materializer.Dispose();
}                       
```

Take a memory dump before hitting a key, the let the while-loop run for some seconds, and the the second memory dump.
On my system the differential memory dump shows a result as follows:
memoryleak

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes #4659

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus looks like the Akka.Streams.Tests.TCK blew up here

File: src/core/Akka.Streams/ActorMaterializer.cs [hunk 1]
```diff
@@ -304,7 +307,8 @@ public sealed class ActorMaterializerSettings
         public static ActorMaterializerSettings Create(ActorSystem system)
         {
             // need to make sure the default materializer settings are available
-            system.Settings.InjectTopLevelFallback(ActorMaterializer.DefaultConfig());
```
[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/core/Akka.Streams/ActorMaterializer.cs:307]: We have to reject any objects being injected if and only if all of its values are already in the config

[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/core/Akka.Streams/ActorMaterializer.cs:307]: I was hoping that we can avoid that, working on another solution right now

[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/core/Akka.Streams/ActorMaterializer.cs:307]: Yeah I know that's a yak-shaving exercise

**Comment by Aaronontheweb** [comment]
@Arkatufus FYI, the repository Url stuff hasn't been playing nice with the build server https://github.com/akkadotnet/akka.net/pull/4669 - we think it's because we're not running the .NET 5 SDK

**Comment by Arkatufus** [comment]
I see, I guess transition time is always annoying. I've reverted that.
Funny thing that my changes works well locally but missed the mark 4 times magnitude order in the test server

**Comment by Arkatufus** [comment]
Code used to test the memory leak (Ommited in PR). Heap memory are sampled on the break points:
image

Heap memory snapshot on breakpoint 1:
image

Heap memory snapshot on breakpoint 2:
image

About 500 new objects are created and 22KB of memory consumed after 5000 iteration

**Comment by Aaronontheweb** [comment]
@Arkatufus you're going to submit another set of screenshots showing the "after" results from the fix too, right?

**Comment by Arkatufus** [comment]
Oh, those are the after fix screenshot, I'll make the before screenshot.

**Comment by Arkatufus** [comment]
I can't even run the test code for the before case, it just froze due to GC cycles and the memory pegged at 130 MiB. Here is the result for a 500 iteration instead.

Heap memory snapshot at breakpoint 1:
image

Heap memory snapshot at breakpoint 2 (after 500 iterations):
image

Over 140,000 new objects generated, using about 6 Mb memory:
image

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**